pkgs/misc/vscode-extensions/vscodeWithConfiguration.nix

```diff
@@ -0,0 +1,39 @@
+# wrapper over vscode to control extensions per project (extensions folder will be created in execution path)
+{ pkgs                            ? import <nixpkgs> {}
+, lib                             ? pkgs.lib
+, writeShellScriptBin             ? pkgs.writeShellScriptBin
+, extensionsFromVscodeMarketplace ? pkgs.vscode-utils.extensionsFromVscodeMarketplace
+
+##User input
+
+, vscode                          ? pkgs.vscode
+# extensions to be symlinked into the project's extensions folder
+, nixExtensions                   ? []
+# extensions to be copied into the project's extensions folder
+, mutableExtensions               ? []
+, vscodeExtsFolderName            ? ".vscode-exts"
+}:
+let 
+  nixExtsDrvs = extensionsFromVscodeMarketplace nixExtensions;
+  mutExtsDrvs = extensionsFromVscodeMarketplace mutableExtensions;
+
+  #removed not defined extensions
+  rmExtensions =  lib.optionalString (nixExtensions++mutableExtensions != []) ''
+    find ${vscodeExtsFolderName} -mindepth 1 -maxdepth 1 ${lib.concatMapStringsSep " " (e : ''! -iname ${e.publisher}.${e.name}'') (nixExtensions++mutableExtensions)} -exec sudo rm -rf {} \;
```

Comment:
I don't really know how I feel about the fact `sudo` has to be used here.

Reply:
yes I don't like it too, any better idea for syncing the extensions folder (we need to be consistent with the nix expression when vscode starts) ?